src/Common/filesystemHelpers.cpp

```diff
@@ -218,23 +218,16 @@ String getFilesystemName([[maybe_unused]] const String & mount_point)
 
 bool pathStartsWith(const std::filesystem::path & path, const std::filesystem::path & prefix_path)
 {
-    String absolute_path = std::filesystem::weakly_canonical(path);
-    String absolute_prefix_path = std::filesystem::weakly_canonical(prefix_path);
-    return absolute_path.starts_with(absolute_prefix_path);
+    std::error_code ec;
+    if (fs::equivalent(path, prefix_path, ec))
+        return true;
+    auto rel = std::filesystem::relative(path, prefix_path);
+    return (!rel.empty() && rel.native()[0] != '.');
 }
 
 static bool fileOrSymlinkPathStartsWith(const std::filesystem::path & path, const std::filesystem::path & prefix_path)
```

Comment:
It seems there any reason for having two functions `pathStartsWith` and `fileOrSymlinkPathStartsWith` now.